wallet/wallet.c

```diff
@@ -3366,6 +3367,32 @@ wallet_payment_list(const tal_t *ctx,
 						  " WHERE payment_hash = ?"
 						  " ORDER BY id;"));
 		db_bind_sha256(stmt, 0, payment_hash);
+	} else if (status) {
+		// TODO(vincenzopalazzo): Missing the filter options are both not null
+		// A possible solution is divided the string in two part (pre-where and post-where) and
+		// use a small if else to set the remain string, with this method we can have small code.
```

Comment:
That would not work with our prepared statement + SQL dialect specialization. Better to have a generic query like this:

`SELECT * FROM payments WHERE (status = ? OR ?) AND (payment_hash = ? OR ?)`

This allows us to disable either restriction by passing `0` or `1` depending on whether we want to filter by that clause or not.

Reply:
So the bindings look like this:

 - `NULL, 1, NULL, 1` for no filtering
 - `complete, 0, NULL, 1` for filtering by `status`
 - `NULL, 1, aaaaaaaa..., 0` for filtering by `payment_hash`
 - `complete, 0, aaaaaaa., 0` for combined filtering by both
 
Prevents dynamic query building, while still enabling full flexibility.

Reply:
It feedback was amazing, thank @cdecker!